examples/date/handler.go

```diff
@@ -124,22 +125,31 @@ func (h *moqHandler) getHandler() moqtransport.Handler {
 				log.Printf("failed to accept subscription: %v", err)
 				return
 			}
-			publisher, ok := w.(moqtransport.Publisher)
+			p, ok := w.(moqtransport.Publisher)
 			if !ok {
 				log.Printf("subscription response writer does not implement publisher?")
 			}
+			datePublisher := &publisher{
+				p:          p,
+				sessionID:  sessionID,
+				subscribID: r.SubscribeID,
+				trackAlias: r.TrackAlias,
+				session:    session,
+			}
 			h.lock.Lock()
-			h.publishers[publisher] = struct{}{}
+			h.publishers[datePublisher] = struct{}{}
 			h.lock.Unlock()
 		}
 	})
 }
 
 func (h *moqHandler) handle(conn moqtransport.Connection) {
+	id := h.nextSessionID
+	h.nextSessionID++
```

Comment:
I think this should be an atomic counter, at least in principle.

Reply:
Good point, thanks!